istioctl/cmd/istioctl/inject.go

```diff
@@ -252,21 +253,18 @@ istioctl kube-inject -f deployment.yaml -o deployment-injected.yaml --injectConf
 			}
 
 			var sidecarTemplate string
-			if injectConfigFile != "" {
-				injectionConfig, err := ioutil.ReadFile(injectConfigFile) // nolint: vetshadow
-				if err != nil {
-					return err
-				}
-				var config inject.Config
-				if err := yaml.Unmarshal(injectionConfig, &config); err != nil {
-					return err
-				}
-				sidecarTemplate = config.Template
-			} else if injectConfigMapName != "" {
-				if sidecarTemplate, err = getInjectConfigFromConfigMap(kubeconfig); err != nil {
-					return err
+
+			// hub and tag params only work with ISTIOCTL_USE_BUILTIN_DEFAULTS
+			// so must be specified together. hub and tag no longer have defaults.
+			if hub != "" || tag != "" {
```

Comment:
This should  be `&&` instead of `||` so we avoid using an empty string.

Reply:
I will add that check inside the block and fail for deprecated BUILTIN behaviour.